Library/Client.ts

```diff
@@ -234,10 +234,11 @@ class Client {
      */
     public track(
         data: ContractsModule.Contracts.Data<ContractsModule.Contracts.Domain>,
-        tagOverrides?: { [key: string]: string; }) {
+        tagOverrides?: { [key: string]: string; },
+        contextObject?: any) {
```

Comment:
I share Alex's concern. The type of the context needs to be documented in some form to enable users to write code against it with some assurance that the context type won't change in a breaking way. Maybe union with `any` to keep it flexible, like: `ITelemetryContext | any` ?
```typescript
interface ITelemetryContext { ... }
interface ITelemetryRequestContext extends ITelemetryContext {
  request: http.ClientRequest;
  response: ...
}
```